workflow/controller/scope.go

```diff
@@ -68,13 +72,56 @@ func (s *wfScope) resolveParameter(v string) (string, error) {
 }
 
 func (s *wfScope) resolveArtifact(v string) (*wfv1.Artifact, error) {
-	val, err := s.resolveVar(v)
+
+	artRef := strings.SplitN(v, "/", 2)
+
+	val, err := s.resolveVar(artRef[0])
+
 	if err != nil {
 		return nil, err
 	}
 	valArt, ok := val.(wfv1.Artifact)
 	if !ok {
 		return nil, errors.Errorf(errors.CodeBadRequest, "Variable {{%s}} is not an artifact", v)
 	}
+
+	if len(artRef) == 2 {
+		fstTmpl := fasttemplate.New(artRef[1], "{{", "}}")
+		subPath, err := common.Replace(fstTmpl, s.getParameters(), true)
+		if err != nil {
+			return nil, err
+		}
+
+		// Copy resolved artifact pointer before adding subpath
+		copyArt := valArt.DeepCopy()
+
+		if copyArt.S3 != nil {
```

Comment:
this should probably be a switch on `copyArt.GetType()`

Reply:
Addressed in 60a757c0e9c314e01a59977471390ef9375f62a2 and 2cc585ede7f1ec5d2b781e4e83caaf1d05bfb803